lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -140,7 +140,30 @@ public function get_items( $request ) {
 			$response[] = $this->prepare_response_for_collection( $data );
 		}
 
-		return rest_ensure_response( $response );
+		$response = rest_ensure_response( $response );
+		unset( $prepared_args['number'] );
+		unset( $prepared_args['offset'] );
+		$total_terms = wp_count_terms( $this->taxonomy, $prepared_args );
+		$response->header( 'X-WP-Total', (int) $total_terms );
```

Comment:
I'd like to implement as headers for now, as it follows the existing implementation, and we can reformat at a later date in this milestone.